### PR TITLE
[dtensor][debug] allow visualize_sharding to print header

### PR DESCRIPTION
**Pull request by XilunWu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #120265
* #121217
* #121216
* #120260
* __->__ #121179

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

### CODE REVIEW

File: torch/distributed/_tensor/debug/visualize_sharding.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def compute_local_shape_and_global_offset(
         return tuple(local_shape), tuple(global_offset)
 
 
-def visualize_sharding(dtensor):
+def visualize_sharding(dtensor, prompt=""):
```
Comment (wanchaol) [inline torch/distributed/_tensor/debug/visualize_sharding.py:133]: maybe just call it header?

File: torch/distributed/_tensor/debug/visualize_sharding.py [hunk 2]
```diff
@@ -154,4 +154,5 @@ def visualize_sharding(dtensor):
     # Convert offsets to blocks with row_ranges for tabulate
     blocks = _convert_offset_to_ranges(all_offsets)
     if device_mesh.get_rank() == 0:
+        print(prompt)
```
Comment (wanchaol) [inline torch/distributed/_tensor/debug/visualize_sharding.py:157]: I think in addition to printing the header, I wonder if we should print some metadata first before printing the actual visualization, i.e. sth like:
dtensor shape: (...), device mesh: (...), placements: (...)

**Review by wanchaol — APPROVED** [review]
lgtm

File: torch/distributed/_tensor/debug/visualize_sharding.py (hunk 2 above)
Comment (XilunWu) [reply torch/distributed/_tensor/debug/visualize_sharding.py:157]: would that be optional to user? 
```
def visualize_sharding(..., print_metadata=False):
    ...
    if print_metadata:
        print(shape)
        print(mesh)
        ...
```

File: torch/distributed/_tensor/debug/visualize_sharding.py (hunk 1 above)
Comment (XilunWu) [reply torch/distributed/_tensor/debug/visualize_sharding.py:133]: yeah that name is appropriate

**Pull request closed without merging**